cmd/cluster-agent/app/app.go

```diff
@@ -157,6 +158,26 @@ func start(cmd *cobra.Command, args []string) error {
 		log.Warnf("Can't initiliaze the runtime settings: %v", err)
 	}
 
+	// Setup Internal Profiling
+	if v := config.Datadog.GetInt("internal_profiling.block_profile_rate"); v > 0 {
+		if err := settings.SetRuntimeSetting("runtime_block_profile_rate", v); err != nil {
+			log.Errorf("Error setting block profile rate: %v", err)
+		}
+	}
+
+	if v := config.Datadog.GetInt("internal_profiling.mutex_profile_fraction"); v > 0 {
+		if err := settings.SetRuntimeSetting("runtime_mutex_profile_fraction", v); err != nil {
+			log.Errorf("Error mutex profile fraction: %v", err)
+		}
+	}
+
+	if config.Datadog.GetBool("internal_profiling.enabled") {
+		err := settings.SetRuntimeSetting("internal_profiling", true)
+		if err != nil {
+			log.Errorf("Error starting profiler: %v", err)
+		}
+	}
+
```

Comment:
I'm wondering if we should create a shared function with other agents.

Reply:
we have it in several place, but never all the same options....
* https://github.com/DataDog/datadog-agent/blob/9925e508eec861aa809668e2dcf145bacda27174/cmd/system-probe/config/config.go#L132
* https://github.com/DataDog/datadog-agent/blob/45c01f4bab71f397497edaee2f662712eabd04b0/cmd/trace-agent/run.go#L276
* https://github.com/DataDog/datadog-agent/blob/92d7c61fa33ecebc4575027f0b20cf630fcf029e/cmd/process-agent/main_common.go#L219

Reply:
I added a shared function between the cluster agent and the core agent.  But process/trace/system-probe are building their own profiling config so it doesn't make sense to include them imho.